### PR TITLE
[sil-opaque-values] OwnershipVerifier support.

### PR DESCRIPTION
**Pull request by atrick**

### CODE REVIEW

**Comment by atrick** [comment]
@swift-ci smoke test.

File: lib/SIL/SILOwnershipVerifier.cpp [hunk 1]
```diff
@@ -999,64 +1001,45 @@ OwnershipCompatibilityUseChecker::visitApplyArgument(ValueOwnershipKind Kind,
...
     return {true, false};
-  case SILArgumentConvention::Direct_Guaranteed:
-    return visitApplyArgument(ValueOwnershipKind::Guaranteed, false);
-  case SILArgumentConvention::Direct_Deallocating:
-    llvm_unreachable("No ownership associated with deallocating");
-  }
+  case ParameterConvention::Indirect_In_Guaranteed:
+  case ParameterConvention::Direct_Guaranteed:
+    return visitApplyParameter(ValueOwnershipKind::Guaranteed, false);
+  default:
```
Comment (gottesmm) [inline lib/SIL/SILOwnershipVerifier.cpp]: Can you keep this as a covered switch?

File: test/SIL/ownership-verifier/opaque_use_verifier.sil [hunk 2]
```diff
@@ -14,3 +14,19 @@ bb0(%0 : @trivial $Builtin.Int32):
...
   return %1 : $T
 }
+
+sil @opaque_identity : $@convention(thin) <T> (@in T) -> @out T {
+bb0(%0 : @owned $T):
+  return %0 : $T
+}
+
+sil @opaque_arg_return : $@convention(thin) <T> (@in T) -> @out T {
+bb0(%0 : @owned $T):
```
Comment (gottesmm) [inline test/SIL/ownership-verifier/opaque_use_verifier.sil]: Can you add a guaranteed test case?

**Review by gottesmm — COMMENTED** [review]
Two requests. But overall looks good.

File: lib/SIL/SILOwnershipVerifier.cpp (hunk 1 above)
Comment (atrick) [inline lib/SIL/SILOwnershipVerifier.cpp]: +1

File: test/SIL/ownership-verifier/opaque_use_verifier.sil (hunk 2 above)
Comment (atrick) [inline test/SIL/ownership-verifier/opaque_use_verifier.sil]: Yeah, I was looking at that case next.

**Comment by atrick** [comment]
Oops. Accidentally reused the same branch. Closing.

**Pull request closed without merging**